applets/cosmic-applet-power/src/main.rs

```diff
@@ -1,62 +1,295 @@
-                        #[wrap(Some)]
-                        set_child = &Label {
-                            set_label: "Settings...",
-                            set_halign: Align::Start,
-                            set_hexpand: true
-                        },
-                        connect_clicked => move |_| {
-                            let _ = Command::new("cosmic-settings").spawn();
-                        }
-                    },
-                    append = &Separator {},
-                    append: &ui::session::build(),
-                    append: second_separator = &Separator {},
-                    append: &ui::system::build(),
+
+use crate::cosmic_session::CosmicSessionProxy;
+use crate::session_manager::SessionManagerProxy;
+
+pub fn main() -> cosmic::iced::Result {
+    let helper = CosmicAppletHelper::default();
+    Audio::run(helper.window_settings())
+}
+
+#[derive(Default)]
+struct Audio {
+    applet_helper: CosmicAppletHelper,
+    icon_name: String,
+    theme: Theme,
+    popup: Option<window::Id>,
+    id_ctr: u32,
+}
+
+#[derive(Debug, Clone)]
+enum Message {
+    Lock,
+    LogOut,
+    Suspend,
+    Restart,
+    Shutdown,
+    TogglePopup,
+    Settings,
+    Ignore,
+    Zbus(Result<(), zbus::Error>),
+}
+
+impl Application for Audio {
+    type Message = Message;
+    type Theme = Theme;
+    type Executor = executor::Default;
+    type Flags = ();
+
+    fn new(_flags: ()) -> (Audio, Command<Message>) {
+        (
+            Audio {
+                icon_name: "system-shutdown-symbolic".to_string(),
+                ..Default::default()
+            },
+            Command::none(),
+        )
+    }
+
+    fn title(&self) -> String {
+        String::from("Power")
+    }
+
+    fn theme(&self) -> Theme {
+        self.theme
+    }
+
+    fn close_requested(&self, _id: iced_sctk::application::SurfaceIdWrapper) -> Self::Message {
+        Message::Ignore
+    }
+
+    fn style(&self) -> <Self::Theme as application::StyleSheet>::Style {
+        <Self::Theme as application::StyleSheet>::Style::Custom(|theme| Appearance {
+            background_color: Color::from_rgba(0.0, 0.0, 0.0, 0.0),
+            text_color: theme.cosmic().on_bg_color().into(),
+        })
+    }
+
+    fn subscription(&self) -> Subscription<Message> {
+        Subscription::none()
+    }
+
+    fn update(&mut self, message: Message) -> Command<Message> {
+        match message {
+            Message::TogglePopup => {
+                if let Some(p) = self.popup.take() {
+                    destroy_popup(p)
+                } else {
+                    self.id_ctr += 1;
+                    let new_id = window::Id::new(self.id_ctr);
+                    self.popup.replace(new_id);
+
+                    let popup_settings = self.applet_helper.get_popup_settings(
+                        window::Id::new(0),
+                        new_id,
+                        (400, 300),
+                        Some(200),
```

Comment:
I think you left in the x-offset that you used while testing? Otherwise looks good!

Reply:
Good catch. Fixed in 9becbba404250b56a6d61b8604f8e6054863a139